Client/Frontend/Browser/Playlist/Cache/PlaylistCacheLoader.swift

```diff
@@ -0,0 +1,466 @@
+        fileExtension = nil
+    }
+    
+    private func findHeader(offset: Int, data: Data, header: [UInt8]) -> Bool {
+        if offset < 0 || data.count < offset + header.count {
+            return false
+        }
+        
+        return [UInt8](data[offset..<(offset + header.count)]) == header
+    }
+    
+    /// Converts a File Extension to a Mime-Type
+    private static func fileExtensionToMimeType(_ fileExtension: String) -> String? {
+        if #available(iOS 14.0, *) {
+            return UTType(tag: fileExtension, tagClass: .filenameExtension, conformingTo: nil)?.preferredMIMEType
+        } else {
+            if let tag = UTTypeCreatePreferredIdentifierForTag(kUTTagClassFilenameExtension, fileExtension as CFString, nil)?.takeRetainedValue() {
+                return UTTypeCopyPreferredTagWithClass(tag, kUTTagClassMIMEType)?.takeRetainedValue() as String?
+            }
+            return nil
+        }
+    }
+    
+    /// Converts a Mime-Type to File Extension
+    private static func mimeTypeToFileExtension(_ mimeType: String) -> String? {
+        if #available(iOS 14.0, *) {
+            return UTType(tag: mimeType, tagClass: .mimeType, conformingTo: nil)?.preferredFilenameExtension
+        } else {
+            if let tag = UTTypeCreatePreferredIdentifierForTag(kUTTagClassMIMEType, mimeType as CFString, nil)?.takeRetainedValue() {
+                return UTTypeCopyPreferredTagWithClass(tag, kUTTagClassFilenameExtension)?.takeRetainedValue() as String?
+            }
+            return nil
+        }
+    }
+    
+    /// Converts a list of AVFileType to a list of file extensions
+    private static func supportedAVAssetFileExtensions() -> [String] {
+        if #available(iOS 14.0, *) {
+            let types = AVURLAsset.audiovisualTypes()
+            return types.compactMap({ UTType($0.rawValue)?.preferredFilenameExtension }).filter({ !$0.isEmpty })
+        } else {
+            let types = AVURLAsset.audiovisualTypes()
+            return types.compactMap({ UTTypeCopyPreferredTagWithClass($0 as CFString, kUTTagClassFilenameExtension)?.takeRetainedValue() as String? }).filter({ !$0.isEmpty })
+        }
+    }
+    
+    /// Converts a list of AVFileType to a list of mime-types
+    private static func supportedAVAssetMimeTypes() -> [String] {
+        if #available(iOS 14.0, *) {
+            let types = AVURLAsset.audiovisualTypes()
+            return types.compactMap({ UTType($0.rawValue)?.preferredMIMEType }).filter({ !$0.isEmpty })
+        } else {
+            let types = AVURLAsset.audiovisualTypes()
+            return types.compactMap({ UTTypeCopyPreferredTagWithClass($0 as CFString, kUTTagClassMIMEType)?.takeRetainedValue() as String? }).filter({ !$0.isEmpty })
+        }
+    }
+    
+    private let knownFileExtensions = [
+        "mov",
+        "qt",
+        "mp4",
+        "m4v",
+        "m4a",
+        "m4b", // DRM protected
```

Comment:
Are we able to cache and play these extensions?

Reply:
Yes, these are a list of all files that can be played. They are currently cached here.